src/config.rs

```diff
@@ -242,11 +242,11 @@ pub struct Cfg {
     pub env_override: Option<String>,
     pub dist_root_url: String,
     pub dist_root_server: String,
-    pub notify_handler: Arc<dyn Fn(Notification<'_>)>,
+    pub(crate) notify_handler: Arc<dyn Fn(Notification<'_>)>,
```

Comment:
I don't recall if we decided this works right now?

Long term, I'd prefer to get rid of the pub fields entirely and switch to accessor methods, but that's a much larger piece of work.

Reply:
@rbtcollins mentioned here https://github.com/rust-lang/rustup/pull/2735#discussion_r642432379 that we need to keep pub instead of using pub(crate) and I modified it.